parameter/ParameterType.cpp

```diff
@@ -151,12 +151,12 @@ void CParameterType::signExtend(int64_t& iData) const
 template <typename type>
 void CParameterType::doSignExtend(type& data) const
 {
-    uint32_t uiSizeInBits = getSize() * 8;
-    uint32_t uiShift = 8 * sizeof(data) - uiSizeInBits;
+    size_t sizeInBits = getSize() * 8;
+    size_t shift = 8 * sizeof(data) - sizeInBits;
```

Comment:
`shift = (sizeof(type) - getSize()) * CHAR_BIT;` (not related to this patch, though).